### ISSUE TITLE
[sival,sensor_ctrl] Fix unexpected alert 5

### ISSUE THREAD

**Issue by matutem** [lowRISC/opentitan#20798]
### Description

Sensor_ctrl is generating spurious alert at bit 5, which can cause failures in any test that examines SENSOR_CTRL RECOV_ALERT expecting specific bits to be on.

One such test is pwrmgr_random_sleep_all_wake_ups, which for sensor_ctrl wakeup the test sets ALERT_TRIGGER bit 0 to trigger a recoverable alert.

Upon wakeup it clears bit 0 for both ALERT_TRIGGER and RECOV_ALERT CSRs, but that doesn't clear the pwrmgr's WAKE_STATUS CSR because RECOV_ALERT bit 5 is still on.

Once this bit is also cleared WAKE_STATUS is finally clear and the test passes. We need clarification from the AST owner that this behavior is as expected.

In order to test the behavior of SENSOR_CTRL, or once this extra alert is not generated, comment out the code that clears bit 5 in //sw/device/tests/sim_dv/rstmgr_alert_info_test.c, indicated by a TODO referring to lowrisc/opentitan#20798.

**Comment by msfschaffner** [comment]
We should probably add a masking mechanism for AST alerts to `sensor_ctrl` as a fall back. 
That way flaky alerts can be stopped from creating spurious wakeup requests and/or alert events.

**Comment by matutem** [comment]
So a new 11 bit wide alert_en sort of CSR? I expect the logic should only disable recoverable events. It probably should also be locked by cfg_regwen.

**Comment by msfschaffner** [comment]
We can discuss adding it on the OT side. Nevertheless, the issue should be root-caused on the AST side as well...

**Comment by meisnere** [comment]
Let me rephrase it, the issue will be fixed on ast side on next TO.
I want to understand what is the suggested fix on this thread, aren't we double fixing the issue? ast &ot side?

**Comment by msfschaffner** [comment]
Ok great that is what I wanted to know.

I think the proposal is to:
- fix the actual issue on the AST side
- make the OT side more generic so that any future issues that are similar could be masked with a generic masking mechanism that is unrelated to the specific nature of this issue.

Does that make sense?

**Comment by meisnere** [comment]
It make sense if it doesn't require ast changes, meaning the described extra masking will be an ot feature while handling false alerts from ast (even though there shouldn't be any on next TO)

**Comment by msfschaffner** [comment]
This as been addressed in open source via https://github.com/lowRISC/opentitan/pull/21870.
